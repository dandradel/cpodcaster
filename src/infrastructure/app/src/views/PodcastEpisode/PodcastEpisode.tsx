import { Box } from '@mui/material'
import React, { ReactElement } from 'react'
import { useLocation } from 'react-router-dom'
import Episode from '../../components/Episode/Episode'
import Error from '../../components/Error/Error'
import PodcastCard from '../../components/PodcastCard/PodcastCard'
import '../../components/PodcastCover/PodcastCover.styles.css'
import Skeleton from '../../components/Skeleton/Skeleton'
import useError from '../../shared/hooks/useError'
import usePodcastById from '../../shared/hooks/usePodcastById'
import { podcastDetailBoxStyles } from '../utils'

function PodcastEpisode(): ReactElement {
  const { podcastCardProps } = usePodcastById()
  const { error } = useError()
  const { state: { episode = {} } = { episode: {} } } = useLocation()

  if (!podcastCardProps || !episode) {
    return <Skeleton />
  }

  if (error || !episode) {
    return <Error />
  }

  return (
    <>
      <Box sx={podcastDetailBoxStyles}>
        <PodcastCard podcastCard={podcastCardProps} />
        <Episode episode={episode} />
      </Box>
    </>
  )
}

export default PodcastEpisode
