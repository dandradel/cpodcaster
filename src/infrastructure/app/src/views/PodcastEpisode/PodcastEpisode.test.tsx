import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, renderWithProviders, screen } from '../../shared/test/test-utils'
import PodcastEpisode from './PodcastEpisode'
import episodeMock from '../../shared/__mocks__/episodes.json'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    state: {
      episode: episodeMock[0],
    },
  }),
}))

jest.mock('../../shared/hooks/usePodcastById', () => ({
  __esModule: true,
  default: () => ({
    podcastCardProps: {
      podcastImage: '1',
      podcastName: 'test',
      podcastAuthor: 'test',
      podcastDescription: 'test',
    },
  }),
}))

describe('PodcastEpisode', () => {
  afterEach(cleanup)

  test('renders PodcastEpisode', () => {
    const container = renderWithProviders(<PodcastEpisode />)
    expect(container).toMatchSnapshot()
  })

  test('renders PodcastEpisode with PodcastCard', () => {
    renderWithProviders(<PodcastEpisode />)
    const podcastCard = screen.getByTestId('podcast-wrapper-testid')
    expect(podcastCard).toBeInTheDocument()
  })

  test('renders PodcastEpisode with Episode', () => {
    renderWithProviders(<PodcastEpisode />)
    const episode = screen.getByTestId('episode-wrapper-testid')
    expect(episode).toBeInTheDocument()
  })
})
