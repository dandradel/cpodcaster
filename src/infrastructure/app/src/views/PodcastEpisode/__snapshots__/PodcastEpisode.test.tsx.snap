// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastEpisode renders PodcastEpisode 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-70hkv0"
      >
        <div
          class="MuiBox-root css-6m2utx"
          data-testid="podcast-wrapper-testid"
        >
          <div
            class="MuiBox-root css-1ia54b9"
          >
            <img
              class="podcastCard__image"
              data-testid="podcast-image-testid"
              src="1"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-1d8isoc"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
              data-testid="podcast-name-testid"
            >
              test
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
              data-testid="podcast-author-testid"
            >
              by 
              test
            </h6>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-tcr4li"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
              data-testid="podcast-description-testid"
            >
              Description:
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
              data-testid="podcast-description1-testid"
            >
              test
            </h6>
          </div>
        </div>
        <div
          class="MuiBox-root css-y6c8j4"
          data-testid="episode-wrapper-testid"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            data-testid="episode-name-testid"
          >
            Episode 1
          </h5>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kr9wzm-MuiTypography-root"
            data-testid="episode-description-testid"
          >
            Description 1
          </h6>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1sjh9xo-MuiDivider-root"
          />
          <audio
            controls=""
            data-testid="audio-podcast-testid"
            style="width: 100%; margin-top: 1rem;"
          >
            <source
              src="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3"
              type="string/string"
            />
          </audio>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-70hkv0"
    >
      <div
        class="MuiBox-root css-6m2utx"
        data-testid="podcast-wrapper-testid"
      >
        <div
          class="MuiBox-root css-1ia54b9"
        >
          <img
            class="podcastCard__image"
            data-testid="podcast-image-testid"
            src="1"
          />
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1d8isoc"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
            data-testid="podcast-name-testid"
          >
            test
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
            data-testid="podcast-author-testid"
          >
            by 
            test
          </h6>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-tcr4li"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
            data-testid="podcast-description-testid"
          >
            Description:
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
            data-testid="podcast-description1-testid"
          >
            test
          </h6>
        </div>
      </div>
      <div
        class="MuiBox-root css-y6c8j4"
        data-testid="episode-wrapper-testid"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          data-testid="episode-name-testid"
        >
          Episode 1
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-kr9wzm-MuiTypography-root"
          data-testid="episode-description-testid"
        >
          Description 1
        </h6>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1sjh9xo-MuiDivider-root"
        />
        <audio
          controls=""
          data-testid="audio-podcast-testid"
          style="width: 100%; margin-top: 1rem;"
        >
          <source
            src="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3"
            type="string/string"
          />
        </audio>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
